docs/reduction-in-locks.md

```diff
@@ -0,0 +1,75 @@
+
+* Copy the redo logs: Transfer all redo logs from the checkpoint up to the current Log Sequence Number (LSN) and start tracking new entries.
+
+* Identify and copy all InnoDB tablespace, `.ibd`, files. This step takes most of the time and is performed under the lock.
+
+* Copy non-InnoDB files.
+
+* Gather sync point: Collect synchronization points from all engines (InnoDB LSN, binlog, Global Transaction Identifier (GTID), etc.) by querying the log status.
+
+* Stop the redo follow thread: Cease the redo fllow thread once it has copied up to the sync point.
+
+* Release the backup lock on the server.
+
+If the backup lock is disabled with the [`--lock-ddl=OFF`](./xtrabackup-option-reference.md/#lock-ddl) option, a backup continues while concurrent DDL events are executed. These backups may be invalid and may fail at either the `backup` or the `--prepare` step.
+
+## The `--lock-ddl=REDUCED` option overview
+
+[Percona XtraBackup 8.4.0-2](./release-notes/8.4.0-2.md) adds the [`--lock-ddl=REDUCED`](./xtrabackup-option-reference.md/#lock-ddl) option to reduce the time the server remains under the backup lock. Now, you can execute more `DDL` operations concurrently while the backup is in progress. The backup lock is now taken after copying the `.ibd` files and before copying the `non-InnoDB` files.
+
+Copying the InnoDB files with the `--lock-ddl=REDUCED` option is split into two phases.
+
+### Phase 1: `Operations performed without the lock`
+
+1. Copying all redo logs from checkpoint up to the current LSN and start following new entries.
+2. Starting the redo log thread.
+3. Tracking file operations from the redo log.
+4. Copying of all .ibd without taking any lock.
+5. Acquiring Lock Tables For Backup (LTFB)/Lock Instance For Backup (LIFB). This step ensures no new DDL operations, such as creating or altering tables, will occur.
+6. Quering the log_status to discover the LSN. This step ensures that all subsequent operations are tracked and that the backup process captures a consistent state of the database.
+7. Copying all InnoDB files.
+8. Ensuring the redo log has catch up to LSN from step 5.
+
+### Phase 2: `Operations performed under the lock`
+
+9. Checking the file operations that were tracked and recopying the tablespaces.
+10. Creating additional `meta` files to perform the required actions (deletions or renames) on the already copied files. This approach ensures that the backup remains consistent and accurate without disrupting the streaming process. This step is required for taking streaming backups.
+11. Gathering a sync point from all engines (InnoDB LSN, binlog, GTID, etc.) by querying the `log_status`.
+12. Stoping the redo follow thread once it copies at least up to sync point at step 11.
+13. Releasing LTFB/LIFB.
+14. Processing the new files during `--prepare` phase before crash recovery starts.
```

Comment:
Please remove step 14 from under the lock section.
It should be a separate step IMHO. It is executed after backup.

Reply:
Please see Prepare phase changes to handle reduced lock in previous comment

Reply:
updated